### PR TITLE
bug fixes, new features

### PR DESCRIPTION
**Pull request by alexbyk**
Hi. First of all thanks for your work

Changes
1) Fix a bug when inputText was settled twice
2) Fix a bug when passing an empty `[date]` doesn't take effect a second time (source of truth error)
3) Fix a bug when passing an empty `[date]` doesn't clear a date, but changes it to "today" date
4) Add `[disabled]` feature to make the `input` disabled, solves #28
5) Add two way binding `[(date)]` and `onChange` event. `onSelect` fires when a user clicks a date (one way binding), `onChange` fires on every `date` change (2way)
6) Add `[emptyText]` attribute. This text is shown when the `[date]` is falsy

Also there are a few minor changes. You can check them in the `integrated-tests` folder, which I've changed too

!!!Pay attention if you decide to merge this:
1) I deleted generated '.js' and added them to .gitignore. So before publishing this module you should generate them (or automate this process by `package.json.scripts`)
2) [disabled] feature needs more testing. I implemented it only to help resolve this  #28 ticket issue
3) I've updated dependencies
4) I'm going to use this component as a temporary solution, so 'for_origin' is a merging branch, while `master` branch is for my own purposes. If you decide to merge this commit and explore it, use 'for_origin' branch, because master may be rewritten from the scratch by that time
5) Don't blame me if something is broken

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,19 +27,21 @@ import { DatepickerModule as YourAlias } from 'angular2-material-datepicker'
...
+The datepicker component can be called with no arguments. See the [Angular 2 Documentation](https://angular.io/docs/ts/latest/cookbook/component-communication.html) for how to communicate with child components. If you use an event emitter, the datepicker component has an emitter called `onSelect` (when the date is picked), and `dateChange` (for two-way binding).
 
 Optional parameters are listed below.
 
 | Parameter | Type | Description |
 |---|---|---|
 | `accentColor` | string  | Replaces the default blue accent color  |
 |`altInputStyle` | boolean | If `true`, changes the input styling to primarily use the accent color |
+|`disabled` | boolean | If `true`, adds "disabled" attribute to input |
+|`emptyText` | string | This text will be shown if the date is empty |
```
Comment (koleary94) [inline README.md]: thoughts on making this `placeholder` or `placeholderText` instead? I think it's a more standard way of describing text of an empty input.

**Comment by alexbyk** [comment]
ok. Squashed commit even I think it wasn't a good idea) Also changed `emptyText` to `placeholder` (in component and html). So right now before merging you should review the changes carefully)

**Comment by koleary94** [comment]
hey @alexbyk. just tested a lot on my end. looks great 👍 

going to merge

**Comment by alexbyk** [comment]
I'm glad I was helpful.

The next idea you can consider is to replace an `input` with a `button` and remove inline styling. This will make the component more flexible and won't break existing functionality (anyway the `input` right now is readonly and user can't type a text). Also this will make possible to pass some custom code, like icon.

You can examine this feature [here](https://github.com/alexbyk/Angular-2-Datepicker/blob/master/integration-test/app.component.ts#L20-L26) (`<i>` tag is passed) and see how it works here: http://beta.likebb.com/odessa/photos (`<md-icon> is passed`)

I didn't create a pull request for this feature because that's what I needed, maybe your needs are different. But if you find the idea cool, I'll be glad to help (or you can implement it by yourself, it's a simple change)

**Pull request merged**